common/src/main/java/io/pravega/common/util/AbstractBufferView.java

```diff
@@ -118,6 +154,74 @@ public ArrayView readFully(int bufferSize) {
             assert available() == 0;
             return readBuffer;
         }
+        
+
+        @Override
+        public String readUTF() throws UTFDataFormatException {
+            int utflen = readShort() & 0x0000FFFF;
+            char[] chararr = new char[2 * utflen];
+
+            int c, char2, char3;
+            int count = 0;
+            int chararr_count=0;
+
+            ArrayView byteview = readFully(utflen);
```

Comment:
I actually did this initially and it did not work. The trouble is when the short loop at the top for ascii fails and it goes to the main loop it reads the same character again (at the same offset). I couldn't find a way to make it correct without really messing up the loops.
If you see a clean way, let me know.

Reply:
I see, it's because the first while loop reads a character and it will break before advancing the `count` variable, so in the second while loop it will effectively  skip over it. 

1. Could we make `c` an `Integer` (nullable) and set it to null at line 179-180; then check for null at 184?
2. Could we use a boolean `needReading` to do the same (to avoid the `Integer` class)?